Updates to markdown linting

maxwondercorn opened the pull request:
Multiple updates to the markdown linting setup

- Moved selected words from the local dictionary to ember-dictionary
- Updated .remarkignore to lint only 3.8.x onward (2 LTS releases)
- Bumped ember-dictionary version to latest
- Bumped retext-repeated-words version to latest

chancancode commented on .local.dic:
```diff
@@ -136,6 +124,4 @@ unsilence
-VoiceOver
-WAI-ARIA
-WCAG
+VoiceOver
```
Can we have a trailing newline here?

maxwondercorn replied:
Trailing newline added

chancancode commented on .remarkignore:
```diff
@@ -27,4 +27,8 @@ guides/v2.18.0/**/*.md
+guides/v3.4.0/**/*.md
+guides/v3.5.0/**/*.md
+guides/v3.6.0/**/*.md
+guides/v3.7.0/**/*.md
```
Same re: newline

maxwondercorn replied:
Trailing newline added

chancancode commented:
@maxwondercorn do you think you can help with https://github.com/ember-learn/super-rentals-tutorial/issues/13?

That workflow is that we write special "runnable" markdowns in that repo. When we merge a pull request there, CI will build that into "regular" markdown, and integrate the result into this repo, e.g. 
https://github.com/ember-learn/guides-source/pull/1126. Because we don't have linting setup in super-rentals-tutorial, often we will merge a PR over there, only to find out when it's being integrated that it failed some lints (repeated words and what not), and had to fix that in a second PR on super-rentals-tutorial.

It would be nice to "roughly mirror" (i.e. duplicate) the linting setup we used here into super-rentals-tutorial, so the CI there will catch most of the mistakes before we merge it there.

I'm brining it up since you seem to know how these things are setup here 😄

chancancode commented:
@jenweber should this target the octane branch?

jenweber commented:
@chancancode it's ok for this to target `master`. We didn't freeze PRs to 3.13 yet.

maxwondercorn commented:
@chancancode I'll take a look at it.

maxwondercorn commented:
Looks like travis errors were timeouts

chancancode commented:
@maxwondercorn Thank you for working on this! Do you mind rebasing against master?

maxwondercorn commented:
@chancancode I will get the rebase this afternoon

maxwondercorn commented:
Added words from super-rentals-tutorial and rebased against master.  This PR removed some words from the local dictionary and put them into the main ember-dictionary.  

If someone updates their local guides-source copy with a `git pull` they should run `npm install` to pull down the latest dictionary.

chancancode commented:
@maxwondercorn Thank you for working on this (and https://github.com/ember-learn/super-rentals-tutorial/issues/13). Sorry it had taken so long to merge this. The builds are pretty broken right now 😐

chancancode commented:
@maxwondercorn I think the next step is to merge these changes into the octane branch (which is where the new tutorial is hosted). Do you think you have cycles to work on that next?

maxwondercorn commented:
@chancancode Updating the octane branch should be pretty straightforward. Diff the local dictionaries and update the linting dependencies should be all that's required.  I should get to it by the end of the week
